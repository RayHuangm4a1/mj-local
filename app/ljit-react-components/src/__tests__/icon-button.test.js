import React from 'react';
import { shallow, mount, } from 'enzyme';
import IconButton from '../components/icon-button';

const { IconTypeEnums, ColorEnums, SizeEnums, } = IconButton;

jest.mock('antd/lib/icon', () => function mockComponent() {
	return <div />;
});
jest.mock('antd/lib/button', () => function mockComponent() {
	return <div />;
});

describe('IconButton', () => {
	it('should handle default props', () => {
		const {
			type,
			disabled,
			onClick,
		} = IconButton.defaultProps;

		expect(type).toEqual(IconTypeEnums.LEFT);
		expect(disabled).toEqual(false);
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const wrapper = shallow(
			<IconButton
				onClick={() => {}}
				className="mock-class"
				disabled={false}
				type={IconTypeEnums.LEFT}
				color={ColorEnums.PRIVATE}
				size={SizeEnums.LARGE}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by ljit-icon-button class', () => {
		const wrapper = shallow(
			<IconButton
				onClick={() => {}}
				disabled={false}
				type={IconTypeEnums.LEFT}
				color={ColorEnums.PRIVATE}
				size={SizeEnums.LARGE}
			/>
		);

		expect(wrapper.hasClass('ljit-icon-button')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<IconButton
				className={className}
				onClick={() => {}}
				disabled={false}
				type={IconTypeEnums.LEFT}
				color={ColorEnums.PRIVATE}
				size={SizeEnums.LARGE}
			/>
		);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const style = { width: '100px', };
		const disabled = false;
		const type = IconTypeEnums.LEFT;
		const color = ColorEnums.PRIVATE;
		const size = SizeEnums.LARGE;
		const onClick = () => {};
		const wrapper = mount(
			<IconButton
				className={className}
				style={style}
				disabled={disabled}
				onClick={onClick}
				type={type}
				color={color}
				size={size}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().style).toEqual(style);
		expect(wrapper.props().disabled).toEqual(disabled);
		expect(wrapper.props().type).toEqual(type);
		expect(wrapper.props().color).toEqual(color);
		expect(wrapper.props().size).toEqual(size);
		expect(wrapper.props().onClick).toEqual(onClick);
	});

	it('should handle onclick event', () => {
		const onClick = jest.fn();
		const wrapper = shallow(
			<IconButton onClick={onClick}>Button</IconButton>
		);

		wrapper.simulate('click');

		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalled();
	});

	it('should be disabled when pass true', () => {
		const disabled = true;
		const wrapper = mount(
			<IconButton disabled={disabled}>Button</IconButton>
		);

		expect(wrapper.props().disabled).toEqual(true);
	});
});
