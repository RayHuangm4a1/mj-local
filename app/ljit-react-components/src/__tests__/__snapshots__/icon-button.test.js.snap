// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconButton
    className="mock-class"
    disabled={false}
    onClick={[Function]}
    size="large"
    type="left"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        color="default"
        isTranslucent={false}
        size="large"
        spin={false}
        style={Object {}}
        type="left"
      />,
      "className": "ljit-icon-button mock-class",
      "color": "",
      "disabled": false,
      "onClick": [Function],
      "outline": "solid",
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "default",
        "isTranslucent": false,
        "size": "large",
        "spin": false,
        "style": Object {},
        "type": "left",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          color="default"
          isTranslucent={false}
          size="large"
          spin={false}
          style={Object {}}
          type="left"
        />,
        "className": "ljit-icon-button mock-class",
        "color": "",
        "disabled": false,
        "onClick": [Function],
        "outline": "solid",
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "default",
          "isTranslucent": false,
          "size": "large",
          "spin": false,
          "style": Object {},
          "type": "left",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
